### DISCUSSION TITLE
are there a function works as generate_series()of postgresql in datafusion?

### DISCUSSION THREAD

**Discussion by l1t1** [apache/datafusion#9315] [Q&A] [answered]
thanks

**Comment by jonahgao** [comment, marked as answer]
There is an array function with the same name, but it is not a table function like in PostgreSQL.
```sh
DataFusion CLI v36.0.0
❯ select generate_series(1,10);
+-----------------------------+
| range(Int64(1),Int64(10))   |
+-----------------------------+
| [1, 2, 3, 4, 5, 6, 7, 8, 9] |
+-----------------------------+
1 row in set. Query took 0.011 seconds.

❯ select * from generate_series(1,10);
Error during planning: table function 'generate_series' not found
```
You can implement it as a user-defined table function and register it in DataFusion. Here's an [example](https://github.com/apache/arrow-datafusion/blob/main/datafusion-examples/examples/simple_udtf.rs).

**Comment by l1t1** [comment]
could modify the results of generate_series to including the upper bound parameter?
then the same results of postgresql and sqlites?

**Comment by l1t1** [comment]
range() excluding the upper bound
duckdb
```sql
D select generate_series(1,3);
┌───────────────────────┐
│ generate_series(1, 3) │
│        int64[]        │
├───────────────────────┤
│ [1, 2, 3]             │
└───────────────────────┘
D select range(1,3);
┌─────────────┐
│ range(1, 3) │
│   int64[]   │
├─────────────┤
│ [1, 2]      │
└─────────────┘
```

**Comment by l1t1** [comment]
please also add generate_series documents to https://arrow.apache.org/datafusion/user-guide/sql/scalar_functions.html

**Reply by SteveLauC** [reply]
In DataFusion, `range` and `generate_series` seem to be the same thing, and the function is called [range](https://arrow.apache.org/datafusion/user-guide/sql/scalar_functions.html#range), `generate_series` is just an alias.

**Reply by SteveLauC** [reply]
PR [docs: document range() alias generate_series()](https://github.com/apache/arrow-datafusion/pull/9321) filed.

**Comment by l1t1** [comment]
the unnest() function is also needed
```sql
D select unnest(range(1,3));
┌─────────────────────┐
│ unnest(range(1, 3)) │
│        int64        │
├─────────────────────┤
│                   1 │
│                   2 │
└─────────────────────┘
```

**Reply by jonahgao** [reply]
I created PR #9342 to address this issue.
